### PR TITLE
Docs - MiddleManager Affinity "strong" definition

### PR DESCRIPTION
**Pull request by petermarshallio**
Reworded "strong" to emphasise meaning and consequences - OTBO https://the-asf.slack.com/archives/CJ8D1JTB8/p1609558156092800

### Description

Clarify that "strong" is required for full affinity of task assignment.

This PR has:
- [x] been self-reviewed.
- [ ] been tested in a test Druid cluster.

cc: @sthetland @techdocsmith @sixtus

### CODE REVIEW

File: docs/configuration/index.md [hunk 1]
```diff
@@ -1399,7 +1398,7 @@ then the value from the configuration below is used:
 |--------|-----------|-------|
 |`druid.worker.version`|Version identifier for the Indexer.|0|
 |`druid.worker.capacity`|Maximum number of tasks the Indexer can accept.|Number of available processors - 1|
-|`druid.worker.globalIngestionHeapLimitBytes`|Total amount of heap available for ingestion processing. This is applied by automatically setting the `maxBytesInMemory` property on tasks.|60% of configured JVM heap|	
+|`druid.worker.globalIngestionHeapLimitBytes`|Total amount of heap available for ingestion processing. This is applied by automatically setting the `maxBytesInMemory` property on tasks.|60% of configured JVM heap|
```
Comment (asdf2014) [inline docs/configuration/index.md:1401]: nit: the blank character at the end of line is unnecessary :sweat_smile:

**Review by sthetland — APPROVED** [review]
That's a helpful clarification... thanks for the doc change.

File: docs/configuration/index.md [hunk 2]
```diff
@@ -1202,13 +1202,12 @@ Example: a function that sends batch_index_task to workers 10.0.0.1 and 10.0.0.2
 
 ###### Affinity
 
-Affinity configs can be provided to the _equalDistribution_ and _fillCapacity_ strategies using the "affinityConfig"
-field. If not provided, the default is to not use affinity at all.
+Affinity configs can be provided to the _equalDistribution_ and _fillCapacity_ strategies using the "affinityConfig" field. If not provided, the default is to not use affinity at all.
```
Comment (techdocsmith) [inline docs/configuration/index.md:1205]:
```suggestion
Use the `affinityConfig` field to pass affinity configuration to the _equalDistribution_ and _fillCapacity_ strategies. If not provided, the default is to not use affinity at all.
```

File: docs/configuration/index.md [hunk 3]
```diff
@@ -1202,13 +1202,12 @@ Example: a function that sends batch_index_task to workers 10.0.0.1 and 10.0.0.2
...
 
 |Property|Description|Default|
 |--------|-----------|-------|
 |`affinity`|JSON object mapping a datasource String name to a list of indexing service MiddleManager host:port String values. Druid doesn't perform DNS resolution, so the 'host' value must match what is configured on the MiddleManager and what the MiddleManager announces itself as (examine the Overlord logs to see what your MiddleManager announces itself as).|{}|
-|`strong`|With weak affinity (the default), tasks for a dataSource may be assigned to other MiddleManagers if their affinity-mapped MiddleManagers are not able to run all pending tasks in the queue for that dataSource. With strong affinity, tasks for a dataSource will only ever be assigned to their affinity-mapped MiddleManagers, and will wait in the pending queue if necessary.|false|
+|`strong`|A boolean determining whether tasks for a datasource must be assigned to affinity-mapped MiddleManagers, causing tasks to remain queued until a slot is available.  If `false`, tasks for a datasource may be assigned to other MiddleManagers whenever affinity-mapped MiddleManagers are not available to run queued tasks.|false|
```
Comment (techdocsmith) [inline docs/configuration/index.md:1210]:
```suggestion
|`strong`|When `true` tasks for a datasource must be assigned to affinity-mapped MiddleManagers. Tasks remain queued until a slot becomes available.  When `false`, Druid may assign tasks for a datasource to other MiddleManagers when affinity-mapped MiddleManagers are unavailable to run queued tasks.|false|
```

**Review by techdocsmith — COMMENTED** [review]
Some style suggestions.

**Comment by suneet-s** [comment]
I've re-triggered Travis for you @petermarshallio

**Pull request merged**